core/domain/playthrough_issue_registry_test.py

```diff
@@ -18,14 +18,57 @@
 
 from __future__ import annotations
 
+from core import utils
 from core.domain import playthrough_issue_registry
 from core.tests import test_utils
+from extensions.issues.CyclicStateTransitions import CyclicStateTransitions
+from extensions.issues.EarlyQuit import EarlyQuit
+from extensions.issues.MultipleIncorrectSubmissions import (
+    MultipleIncorrectSubmissions)
 
 
 class IssueRegistryUnitTests(test_utils.GenericTestBase):
     """Test for the issue registry."""
 
+    def setUp(self):
+        super(IssueRegistryUnitTests, self).setUp()
+        self.issues_dict = {
+            'EarlyQuit': (
+                EarlyQuit.EarlyQuit),
+            'CyclicStateTransitions': (
+                CyclicStateTransitions.CyclicStateTransitions),
+            'MultipleIncorrectSubmissions': (
+                MultipleIncorrectSubmissions.MultipleIncorrectSubmissions)
+            }
+        self.invalid_issue_type = 'InvalidIssueType'
+
+    def tearDown(self):
+        super(IssueRegistryUnitTests, self).tearDown()
+        playthrough_issue_registry.Registry._issues = {} # pylint: disable=protected-access
+
     def test_issue_registry(self):
         """Do some sanity checks on the issue registry."""
         self.assertEqual(
             len(playthrough_issue_registry.Registry.get_all_issues()), 3)
+
+    def test_correct_issue_registry_types(self):
+        """Tests issue registry for fetching of issue instances of correct
+        issue types.
+        """
+        for issue_type, instance in self.issues_dict.items():
+            self.assertIsInstance(
+                playthrough_issue_registry.Registry.get_issue_by_type(
+                    issue_type), instance)
+
+    def test_incorrect_issue_registry_types(self):
+        """Tests issue registry for raising error on passing incorrect issues
+        types.
+        """
+        with self.assertRaisesRegex(utils.ValidationError, (
+            'Invalid issue type: %s' % self.invalid_issue_type)):
+            try:
```

Comment:
You shouldn't need the try/except in tests. Just call the function and make sure the error is as expected using the assertRaisesRegex method.

Reply:
it is just that there is NO exception catching block in the main file `core/domain/playthrought_issue_registry.py` for get_issue_by_type method and it always throws keyerror when wrong issue_type is passed . However wherever this class is needed necessary methods are implemented there so that no invalid issue_type gets passed down get_issue_by_type . So the only reason i implemented try/except is to make sure that the error is handled in the same way as it is done in other files ex. `core/domain/stats_domain.py` line 840 .

Reply:
This is a test file. You shouldn't have try/except in it (or any significant logic, for that matter). You don't need to make test files behave the same way as the non-test files, they have very different purposes.

Reply:
(Also, in future, if you want me to reply to something, please assign me to the PR, otherwise it may slip.)

Reply:
Thanks Sean . I'll amend the changes asap .